src/nisarqa/utils/lonlat.py

```diff
@@ -2,52 +2,93 @@
 
 import os
 import textwrap
-from collections.abc import Sequence
+from collections.abc import Generator, Iterable, Sequence
 from dataclasses import dataclass
 from pathlib import Path
+from typing import Any
+
+import numpy as np
 
 import nisarqa
 
 objects_to_skip = nisarqa.get_all(__name__)
 
 
-def unwrap_longitudes(lon_lat_points: Sequence[LonLat]) -> list[LonLat]:
+def unwrap_longitudes(lons: Iterable[float]) -> list[float]:
     """
-    Normalize longitudes so that they form a continuous quadrilateral
-    across the antimeridian.
+    Unwrap longitudes if the values cross the antimeridian.
+
+    Specifically, longitudes are normalized so that the absolute difference
+    between any adjacent pair of longitude values is <= 180 degrees.
+
+    Of note, in the case of the an antimeridian crossing, longitude values
+    are "unwrapped" to extend beyond the interval of +/-180 degrees
+    for the crossing.
 
     Arguments
     ---------
-    lon_lat_points : Sequence of nisarqa.LonLat
-        List of nisarqa.LonLat (in degrees)
+    lons : Sequence of float
+        Sequence of longitude values (in degrees).
 
     Returns
     -------
-    unwrapped_lon_lat : list of nisarqa.LonLat
-        Copy of `lon_lat`, but in the case of an antimeridian crossing
-        the longitude values are "unwrapped" to extend beyond the
-        interval of +/-180 degrees. The ordering of the points is preserved.
+    unwrapped : list of nisarqa.LonLat
+        Copy of `lons`, but unwrapped so that the absolute difference
+        between any adjacent pair of longitude values is <= 180 degrees.
+        The ordering of the points is preserved.
     """
-    unwrapped = [lon_lat_points[0]]
 
-    for i in range(1, len(lon_lat_points)):
-        prev_lon = lon_lat_points[i - 1].lon
-        current_lon = lon_lat_points[i].lon
+    # since `lons` is an Iterable, `wrap_to_interval` returns a list
+    lons_360 = nisarqa.wrap_to_interval(val=lons, start=-360, stop=360)
 
-        delta = current_lon - prev_lon
+    unwrapped = [lons_360[0]]
+
+    for prev_lon, curr_lon in nisarqa.pairwise(iterable=lons_360):
+
+        delta = curr_lon - prev_lon
 
         # If it's a large jump to the west, subtract 360
         if delta > 180:
-            current_lon -= 360
+            curr_lon -= 360
         # If it's a large jump to the east, add 360
         elif delta < -180:
-            current_lon += 360
+            curr_lon += 360
 
-        unwrapped.append(LonLat(current_lon, lon_lat_points[i].lat))
+        unwrapped.append(curr_lon)
 
     return unwrapped
 
 
+def normalize_lon_lat_pts(lon_lat_points: Iterable[LonLat]) -> list[LonLat]:
```

Comment:
One thing I don't like about Python is that its type system makes no distinction between `Iterable`s that can be iterated over multiple times (e.g. `list`) vs `Iterable`s that can only be iterated over a single time (e.g. `Generator`).

That's a problem here because we want to iterate over `lon_lat_points` twice -- once to unwrap the longitudes and again to get the latitudes. If `lon_lat_points` was a generator, then the latitudes that we get will be an empty iterable (since we'd already exhausted the generator while unwrapping the longitudes).

In order to avoid this case, it might be prudent to specify that the input is a `Sequence` rather than an `Iterable`, since `Sequence`s can always be iterated over multiple times.

```suggestion
def normalize_lon_lat_pts(lon_lat_points: Sequence[LonLat]) -> list[LonLat]:
```